### PR TITLE
Solution items are now considered when removing empty folders

### LINKED ISSUE #10669: dotnet sln remove puts all nested not empty solution folders into root folder
## Steps to reproduce
1. Create blank solution
2. Create solution folders and any file like this: \NewFolder1\NewFolder2\text.txt
3. Create any project inside the solution
4. Remove project using the "dotnet sln remove" command

## Expected  behavior
File is located in the \NewFolder1\NewFolder2 folder

## Actual behavior
File is located in the \NewFolder2 folder

## Environment data
`dotnet --info` output:
.NET Core SDK (reflecting any global.json):
 Version:   2.1.500
 Commit:    b68b931422

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.17134
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.500\

Host (useful for support):
  Version: 2.1.6
  Commit:  3f4f8eebd8

.NET Core SDKs installed:
  2.1.202 [C:\Program Files\dotnet\sdk]
  2.1.302 [C:\Program Files\dotnet\sdk]
  2.1.400 [C:\Program Files\dotnet\sdk]
  2.1.402 [C:\Program Files\dotnet\sdk]
  2.1.403 [C:\Program Files\dotnet\sdk]
  2.1.500 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.0.9 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

### PR DESCRIPTION
**Pull request by Cosebdd**
Solution items are now considered when removing empty folders. A test added to check whether the logic is maintained. Fixes #10669

https://github.com/dotnet/cli/issues/10669

### CODE REVIEW

**Comment by peterhuene** [comment]
@Cosebdd thank you for your contribution.  I'll review this shortly and provide feedback to you.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/App.sln [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/App.sln]: There's a misspelling in the path.  Should be `SlnFileWithSolutionItemsInNestedFolders`.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/App.config [hunk 2]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/App.config]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/ConsoleApp1.csproj [hunk 3]
```diff
@@ -0,0 +1,53 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/ConsoleApp1.csproj]: I think we can skip adding this file.  It's not needed by the test (`dotnet sln remove` can remove projects from a solution without the project being on disk).

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/Program.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/Program.cs]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/Properties/AssemblyInfo.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp1/Properties/AssemblyInfo.cs]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/App.config [hunk 6]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/App.config]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/ConsoleApp2.csproj [hunk 7]
```diff
@@ -0,0 +1,53 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/ConsoleApp2.csproj]: I think we can skip adding this file.  It's not needed by the test (`dotnet sln remove` can remove projects from a solution without the project being on disk).

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/Program.cs [hunk 8]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/Program.cs]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/Properties/AssemblyInfo.cs [hunk 9]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection;
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/ConsoleApp2/Properties/AssemblyInfo.cs]: I think we can skip adding this file.  It's not needed by the test.

File: TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/TextFile1.txt [hunk 10]
```diff
@@ -0,0 +1 @@
+SomeText
```
Comment (peterhuene) [inline TestAssets/TestProjects/SlnFileWithSolutinItemsInNestedFolders/TextFile1.txt]: I think we can skip adding this file.  It's not needed by the test.

**Comment by peterhuene** [comment]
@Cosebdd it looks like there's an infrastructure issue that caused one of the test runs to fail.  I'll rerun when the tests complete.

**Comment by Cosebdd** [comment]
@peterhuene thanks for reviewing my changes. I've fixed everything you've mentioned. There are some failed checks though. I don't think it's somehow related to my change

**Comment by peterhuene** [comment]
@Cosebdd thanks for the contribution!  As it was too late to get this fix in for the 3.0 release (only very important fixes are being accepted currently), look for it in 5, which will be the next major release of .NET.

**Pull request merged**